test/e2e/framework/util.go

```diff
@@ -3920,6 +3920,30 @@ func WaitForControllerManagerUp() error {
 	return fmt.Errorf("waiting for controller-manager timed out")
 }
 
+// CheckForControllerManagerHealthy checks that the controller manager does not crash within "duration"
+func CheckForControllerManagerHealthy(duration time.Duration) error {
+	var PID string
+	cmd := "sudo docker ps | grep k8s_kube-controller-manager | cut -d ' ' -f 1"
+	for start := time.Now(); time.Since(start) < duration; time.Sleep(5 * time.Second) {
+		result, err := SSH(cmd, GetMasterHost()+":22", TestContext.Provider)
+		if err != nil || result.Code != 0 {
+			// We don't necessarily know that it crashed, pipe could just be broken
+			LogSSHResult(result)
```

Comment:
If the SSH command failed, would that cause the pid check below to fail?

Reply:
Ah yes, I think it would cause stdout to be empty and thus cause the check to fail. I will change the second case to an else if.